implementations/rust/ockam/ockam_command/src/node/create.rs

```diff
@@ -244,76 +134,94 @@ impl CreateCommand {
     }
 }
 
-fn run_background_node(
-    c: CreateCommand,
-    addr: SocketAddr,
-    cfg: OckamConfig,
-    opts: CommandGlobalOpts,
-) -> Result<()> {
-    let (mut ctx, mut executor) = NodeBuilder::without_access_control().no_logging().build();
-
-    executor
-        .execute(async move {
-            let v = run_background_node_impl(&mut ctx, c, addr, cfg, &opts).await;
-
-            match v {
-                Err(e) => {
-                    eprintln!("Background node error {:?}", e);
-                    std::process::exit(1);
-                }
-                Ok(v) => v,
-            }
-        })
-        .map_err(anyhow::Error::from)
+fn run_impl(opts: CommandGlobalOpts, cmd: CreateCommand) -> crate::Result<()> {
+    let verbose = opts.global_args.verbose;
+    let cfg = &opts.config;
+    if cmd.foreground {
+        let cmd = cmd.overwrite_addr()?;
+        let addr = SocketAddr::from_str(&cmd.tcp_listener_address)?;
+        // HACK: try to get the current node dir.  If it doesn't
+        // exist the user PROBABLY started a non-detached node.
+        // Thus we need to create the node dir so that subsequent
+        // calls to it don't fail
+        if cfg.get_node_dir(&cmd.node_name).is_err() {
+            println!("Creating node directory...");
+            cfg.create_node(&cmd.node_name, addr, verbose)?;
+            cfg.persist_config_updates()?;
+        }
+        embedded_node_daemon(run_background_node, (opts.clone(), cmd, addr))?;
```

Comment:
The `run_background_node` mention in this line is strange since we're in the `if cmd.foreground` branch. We should rename that function

Reply:
That's certainly a bad name. We are calling that function only at that line, so we could rename it to `run_foreground_node`